cmd/add/smtpConfiguration/smtpConfiguration.go

```diff
@@ -0,0 +1,199 @@
+		? To Email: <Email ID for Verification>
+		SMTP settings are verified
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return addAccessRestriction()
+		},
+	}
+
+	return cmd
+}
+
+func addAccessRestriction() error {
+	var smtpSchema api.SmtpConfigSchema 
+	var smtpLabels = [] string {"Provider","Key","Secret", "SmtpHost", "SmtpPort",
+	"FromName","FromEmailId","UserName","Password",  "IsSsl"}
+	var smtpProviders []string 
+	var num int
+	var isSsl bool
+	var isMailazy bool 
+	num = len(cmdutil.SmtpProviders)
+	for i := 0; i < num; i++ { 
+		smtpProviders = append(smtpProviders, cmdutil.SmtpProviders[i].Name)
+	}
+	for _,val := range smtpLabels {
+		
+		if val == "Provider" {
+			err := prompt.SurveyAskOne(&survey.Select{
+				Message: cmdutil.SmtpOptionNames[val] + " :",
+				Options:  smtpProviders,
+				}, &num)
+				if err != nil {
+					return err
+				}					
+		} else if val == "IsSsl" && num != 0 {
+			isSsl = cmdutil.SmtpProviders[num].EnableSSL
+			if err := prompt.Confirm(cmdutil.SmtpOptionNames[val] + ":", 
+						&isSsl); err != nil {
+							return err
+			}
+		} else {
+			if num == 0 {
+				isMailazy = ( val != "Password" && val != "IsSsl" && val != "UserName" && val != "SmtpHost" && val != "SmtpPort")
+			} else if num == 9 {
+				isMailazy = (val != "Key" && val != "Secret" )
+			} else if num != 0 {
+				isMailazy = (val != "Key" && val != "Secret" && val != "SmtpHost" && val != "SmtpPort")
+			}
+			if isMailazy == true {
+				configObj := reflect.ValueOf(&smtpSchema).Elem()
+				field := configObj.FieldByName(val)
+				var promptRes string
+				
+				if  val == "Key" {
+					fmt.Println("(If you don't have a mailazy account. Please Create a mailazy account via https://app.mailazy.com/signup)")
+				}
+				prompt.SurveyAskOne(&survey.Input{
+					Message: cmdutil.SmtpOptionNames[val] + ":",
+				}, &promptRes, survey.WithValidator(survey.Required))
+
+				if strings.TrimSpace(promptRes) == "" {
+					return errors.New(cmdutil.SmtpOptionNames[val] + " is required")
+				}
+				if val == "FromEmailId" && !cmdutil.ValidateEmail.MatchString(promptRes)  {
+					return &cmdutil.FlagError{Err: errors.New("Invalid email format")}
+				}
+				if val == "SmtpPort" {
+					smtpPort, err := strconv.ParseInt(promptRes, 10, 64)
+					if err != nil  {
+						return &cmdutil.FlagError{Err: errors.New("Please enter the valid SMTP Port")}
+					}
+					field.SetInt(smtpPort)
+				} else {
+					field.SetString(strings.TrimSpace(promptRes))
+				}
+			}
+		}
+	}
+	if num != 9 {
+		smtpSchema.Provider = smtpProviders[num]
+	}
+	smtpSchema.IsSsl = isSsl
+	
+	if num == 0 {
+      if strings.Contains(conf.HubPageDomain ,"devhub.") {
+        smtpSchema.SmtpHost = "devsmtp.mailazy.com";
+        smtpSchema.SmtpPort = 588;
+      } else {
+		smtpSchema.SmtpHost = "smtp.mailazy.com"
+		smtpSchema.SmtpPort = 587;
+	  }
+	  smtpSchema.UserName = smtpSchema.Key
+	  smtpSchema.Password = smtpSchema.Secret
```

Comment:
We we are assigning, key and secret to Username and password, can't we use it directly

Reply:
This is only for mailazy configuration as we need to pass key, secret ,username and password ENVs and in mailazy , user name and password is same as key and secret